x/auth/ante.go

```diff
@@ -95,7 +95,7 @@ func NewAnteHandler(am AccountKeeper, fck FeeCollectionKeeper) sdk.AnteHandler {
 			fck.AddCollectedFees(newCtx, stdTx.Fee.Amount)
 		}
 
-		for i := 0; i < len(stdSigs); i++ {
+		for i := 0; i < len(stdSigs) && i < ctx.TxSigLimit(); i++ {
```

Comment:
Whoah - this will just not check the remaining signatures!

Rather, what we want to do is:
- Fail if the transaction has too many signatures.
- Check the number of signatures immediately, before we check the validity of any of them.
- Introspect into multisig public-keys and count those as multiple signatures